samcli/hook_packages/terraform/hooks/prepare/resource_linking.py

```diff
@@ -1477,3 +1479,66 @@ def _link_gateway_integration_responses_to_gateway_resource(
         linking_exceptions=exceptions,
     )
     ResourceLinker(resource_linking_pair).link_resources()
+
+
+def _link_gateway_authorizer_to_lambda_function_call_back(
+    gateway_authorizer_cfn_resource: Dict, lambda_function_resource_values: List[ReferenceType]
+) -> None:
+    """
+    Callback function that is used by the linking algorithm to update a CFN Authorizer Resource with
+    a reference to the Lambda function's invocation URI
+
+    Parameters
+    ----------
+    gateway_authorizer_cfn_resource: Dict
+        API Gateway Authorizer CFN resource
+    lambda_function_resource_values: List[ReferenceType]
+        List of referenced Lambda Functions either as the logical id of Lambda Function reosurces
+        defined in the customer project, or ARN values for actual Lambda Functions defined
+        in customer's account. This list should always contain one element only.
+    """
+    if len(lambda_function_resource_values) > 1:
+        raise InvalidResourceLinkingException("Could not link multiple Lambda functions to one Gateway Authorizer")
+
+    logical_id = lambda_function_resource_values[0]
```

Comment:
In theory, `lambda_function_resource_values` will contain something, whether the linking happens with an applied resource, or a non applied resource.

With that being said though, I don't think it would add too much we just coded more defensively since both these checks can be done in a helper method, and applied to all the different callbacks.

Reply:
After some discussion with another team member, a new check will likely happen earlier in the call stack in a different PR to not expand the scope of this PR.

Reply:
Why not change if statement above with `if len(lambda_function_resource_values) != 1:` and also update the exception message?

I do understand that we are adding another check in the flow, but that still leaves a possible error in the future since it will be added in another place. If the execution flow changes or if we call this method from somewhere else then we will get an unexpected exception here.

Reply:
I think we should not raise exception if this list is empty. I see if this list is empty, so it means we could not find any destination resources to be linked to the source resource. It may be invalid semantically, but the purpose of the hook is to map between the TF project and the CFN, and these type of validations are left to the core SAM functionalities.

For example, if the customers want to run `sam local invoke` to test lambda function, but the TF project is not complete, and the customers do not add all the required API Gateway resources, I do not think we should stop them from using SAM CLI.

Reply:
I am ok for raising some exception here in the callback, but not to call the callback function at all if there is no destinations to be linked.